test/jdk/javax/swing/JButton/bug4323121.java

```diff
@@ -30,96 +30,100 @@
+                        windowGainedFocus.countDown();
+                    }
+                });
+
                 frame.pack();
                 frame.setLocationRelativeTo(null);
                 frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
                 frame.setVisible(true);
             });
+
+            if (!windowGainedFocus.await(1, SECONDS)) {
+                throw new RuntimeException("Window didn't gain focus");
+            }
             robot.waitForIdle();
-            robot.delay(1000);
+
             SwingUtilities.invokeAndWait(() -> {
-                pt = button.getLocationOnScreen();
-                buttonW = button.getSize().width;
-                buttonH = button.getSize().height;
+                Point location = button.getLocationOnScreen();
+                buttonCenter = new Point(location.x + button.getWidth() / 2,
+                                         location.y + button.getHeight() / 2);
             });
-            robot.mouseMove(pt.x + buttonW / 2, pt.y + buttonH / 2);
-            robot.waitForIdle();
-            if (failed) {
-                throw new RuntimeException("Any created button returns " +
-                                    "true for isArmed()");
+
+            robot.mouseMove(buttonCenter.x, buttonCenter.y);
+
+            if (!mouseEntered.await(1, SECONDS)) {
+                throw new RuntimeException("Mouse entered event wasn't received");
+            }
+            if (modelArmed) {
+                throw new RuntimeException("getModel().isArmed() returns true "
+                                           + "for a subclass of JButton");
             }
         } finally {
-                SwingUtilities.invokeAndWait(() -> {
+            SwingUtilities.invokeAndWait(() -> {
                 if (frame != null) {
                     frame.dispose();
                 }
             });
         }
     }
 
-    static class testButton extends JButton implements MouseMotionListener, MouseListener {
-        public testButton(String label) {
+    private static final class TestButton extends JButton {
+        public TestButton(String label) {
             super(label);
-            addMouseMotionListener(this);
-            addMouseListener(this);
-        }
-
-        protected void paintComponent(Graphics g) {
-            super.paintComponent(g);
-        }
-
-        protected void paintBorder(Graphics g) {
-        }
-
-        public void mousePressed(MouseEvent e) {
-        }
-
-        public void mouseDragged(MouseEvent e) {
-        }
-
-        public void mouseMoved(MouseEvent e) {
-        }
-
-        public void mouseReleased(MouseEvent e) {
-        }
-
-        public void mouseEntered(MouseEvent e) {
-            if (getModel().isArmed()) {
-                failed = true;
-            }
-        }
-
-        public void mouseExited(MouseEvent e) {
-        }
-
-        public void mouseClicked(MouseEvent e) {
         }
     }
+
```

Comment:
Extra newline

Reply:
I don't get where you wanted to add a blank line…